File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/database/mapper/RealmUserMapper.java
package com.lattis.ellipse.data.database.mapper;

import androidx.annotation.NonNull;

import com.lattis.ellipse.data.database.base.AbstractRealmDataMapper;
import com.lattis.ellipse.data.database.model.RealmUser;
import com.lattis.ellipse.domain.model.User;

import javax.inject.Inject;

public class RealmUserMapper extends AbstractRealmDataMapper<User,RealmUser> {

    @Inject
    public RealmUserMapper() {}

    @NonNull
    @Override
    public RealmUser mapIn(@NonNull User user) {
        RealmUser realmUser = new RealmUser();
        realmUser.setId(user.getId());
        realmUser.setFirstName(user.getFirstName());
        realmUser.setLastName(user.getLastName());
        realmUser.setEmail(user.getEmail());
        realmUser.setPhoneNumber(user.getPhoneNumber());
        realmUser.setRegistrationId(user.getRegistrationId());
        realmUser.setUsingFacebook(user.isUsingFacebook());
        realmUser.setImageURI(user.getImageUri());
        return realmUser;
    }

    @NonNull
    @Override
    public User mapOut(@NonNull RealmUser realmUser) {
        User user = new User();
        user.setId(realmUser.getId());
        user.setFirstName(realmUser.getFirstName());
        user.setLastName(realmUser.getLastName());
        user.setEmail(realmUser.getEmail());
        user.setPhoneNumber(realmUser.getPhoneNumber());
        user.setRegistrationId(realmUser.getRegistrationId());
        user.setUsingFacebook(realmUser.isUsingFacebook());
        user.setImageUri(realmUser.getImageURI());

        return user;
    }
}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/database/mapper/RealmUserMapper.kt
package com.lattis.data.database.mapper

import com.lattis.data.database.base.AbstractRealmDataMapper
import com.lattis.data.database.model.RealmUser
import com.lattis.domain.models.User
import javax.inject.Inject

class RealmUserMapper @Inject constructor() :
    AbstractRealmDataMapper<User, RealmUser>() {
    override fun mapIn(user: User): RealmUser {
        val realmUser = RealmUser()
        realmUser.id = user.id
        realmUser.firstName=user.firstName
        realmUser.lastName = (user.lastName)
        realmUser.email = (user.email)
        realmUser.phoneNumber=(user.phoneNumber)
        realmUser.registrationId =(user.registrationId)
        realmUser.isUsingFacebook = (user.isUsingFacebook)
        realmUser.imageURI=(user.imageUri)
        return realmUser
    }

    override fun mapOut(realmUser: RealmUser): User {
        val user = User()
        user.id = realmUser.id
        user.firstName=realmUser.firstName
        user.lastName = (realmUser.lastName)
        user.email = (realmUser.email)
        user.phoneNumber=(realmUser.phoneNumber)
        user.registrationId =(realmUser.registrationId)
        user.isUsingFacebook = (realmUser.isUsingFacebook)
        user.imageUri= realmUser.imageURI
        return user
    }
}